documentation/release_notes.rst

```diff
@@ -8,6 +8,92 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
+- ``radix_sort`` and ``radix_sort_by_key`` kernel templates residing in
+  the ``oneapi::dpl::experimental::kt::gpu::esimd`` now throw ``std::bad_alloc`` if they fail to allocate global memory.
+- Fixed a potential hanging occurring with ``radix_sort`` and ``radix_sort_by_key`` kernel templates residing in
+  the ``oneapi::dpl::experimental::kt::gpu::esimd`` namespace. 
+- Fixed documentation for ``sort_by_key`` algorithm, which used to be mistakenly described as stable, despite being unstable for host execution policies (``seq``, ``unseq``, ``par``, ``par_unseq``). It may become unstable for device execution policies in the future. If stability is required, use ``stable_sort_by_key`` instead. 
+- Resolved register spills that led to runtime crashes when calling ``sort`` with DPC++ execution policies on CUDA devices.
+
+
+Known Issues and Limitations
+----------------------------
+New in This Release
+^^^^^^^^^^^^^^^^^^^
+- When compiled with -O0 and executed on GPU devices ``histogram`` algorithms may provide incorrect results.
+- Inclusion of ``<oneapi/dpl/dynamic_selection>`` prior to ``<oneapi/dpl/random>`` may result in compilation errors. Include ``<oneapi/dpl/random>`` first as a workaround.
+- Incorrect results may occur when using oneapi::dpl::experimental::philox_engine algorithm with no predefined template parameters with `word_size` values other than 64 and 32.
+- Incorrect results or a synchronous SYCL exception may be observed with DPC++ execution policies when compiling the following algorithms with -O0 optimization: ``exclusive_scan``, ``inclusive_scan``, ``transform_exclusive_scan``, ``transform_inclusive_scan``, ``copy_if``, ``remove``, ``remove_copy``, ``remove_copy_if'``, ``remove_if``, ``partition``, ``partition_copy``, ``stable_partition``, ``unique``, ``unique_copy``, and ``sort``.
+- Calls to ``transform_inclusive_scan``, ``transform_exclusive_scan``, ``inclusive_scan``, and ``exclusive_scan`` using DPC++ execution policies with an initial element require that the value type of the provided input sequence is convertible to the type of the initial element.
+- The following algorithms when called with DPC++ execution policies may apply the user-provided predicate or equality operator for calls with no user predicate ~2N times: ``copy_if``, ``remove``, ``remove_copy``, ``remove_copy_if``, ``remove_if``, ``partition_copy``, ``unique``, and ``unique_copy``.
+- The ``adjacent_find``, ``all_of``, ``any_of``, ``equal``, ``find``, ``find_if``, ``find_end``, ``find_first_of``, ``find_if_not``, ``includes``, ``is_heap``, ``is_heap_until``, ``is_sorted``, ``is_sorted_until``, ``mismatch``, ``none_of``, ``search``, and ``search_n`` algorithms may cause a segmentation fault when used with a DPC++ execution policy for CPU device, and built on Linux with Intel® oneAPI DPC++/C++ Compiler 2025.0.0 and -O0 -g compiler options.
+
```

Comment:
```suggestion
``sort``, ``stable_sort`` algorithms residing in ``oneapi::dpl::ranges`` namespace
internally work with ``std::swap`` instead of ``std::ranges::iter_swap`` for swapping elements.
That is, customizations targeting ``std::ranges::iter_swap`` will not be respected.

```

@akukanov, could you correct my suggestion if I am mistaken here?

Reply:
I've made the change.  Please let me know if we need to revert it.

Reply:
So, experiments and search by Dmitry and myself indicate that the C++ implementations of `std::ranges::sort` do not use `iter_swap` either. In general, it is not clear if the standard mandates it or not; perhaps not.

Yet I think this is still a useful note, but - since the behavior is not necessarily a bug, and we are not sure if we will be able to improve it - maybe the note should be placed into the library guide and not into release notes.

Reply:
Thanks for the update. I'll remove the line from this PR so we can land the release notes.

Reply:
Done.